okhttp/src/main/java/okhttp3/internal/http/RetryAndFollowUpInterceptor.kt

```diff
@@ -60,6 +60,7 @@ class RetryAndFollowUpInterceptor(private val client: OkHttpClient) : Intercepto
     var followUpCount = 0
     var priorResponse: Response? = null
     var newExchangeFinder = true
+    var recoveredFailures: MutableList<IOException>? = null
```

Comment:
I think you might want to take advantage of Kotlin’s non-mutable List to save allocations in the common case.

This line becomes:

```
     var recoveredFailures = listOf<IOException>() // no allocations
```

And the `appendOrCreateList()` call becomes:

```
  recoveredFailures += e.firstConnectException
```

It’s a little worse performing if there’s a lot of exceptions, but I think the syntax improvement is worthwhile.